urllib3/connectionpool.py

```diff
@@ -592,13 +594,13 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
             release_conn = True
             raise
 
-        except (TimeoutError, HTTPException, SocketError, ConnectionError) as e:
+        except (TimeoutError, HTTPException, SocketError, ProtocolError) as e:
             # Discard the connection for these exceptions. It will be
             # be replaced during the next _get_conn() call.
             conn = conn and conn.close()
             release_conn = True
 
-            if isinstance(e, SocketError) and self.proxy:
+            if isinstance(e, (SocketError, NewConnectionError)) and self.proxy:
```

Comment:
I don't think we can ever hit this if `e` is `NewConnectionError`, can we? I got lost traipsing through the exception heirarchy, but I don't think any of the parent exceptions of `NewConnectionError` are declared in the `except` block.